test/tests/pbs_Rrecord_resources_used.py

```diff
@@ -0,0 +1,616 @@
+            self.skipTest('test requires two MoMs as input, ' +
+                 'use -p moms=<mom1>:<mom2>')
+
+        self.server.set_op_mode(PTL_CLI)
+
+        # PBSTestSuite returns the moms passed in as parameters as dictionary
+        # of hostname and MoM object
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+        self.momA.delete_vnode_defs()
+        self.momB.delete_vnode_defs()
+
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        rc = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA)
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostB)
+        self.assertEqual(rc, 0)
+
+        a = {'resources_available.ncpus': 4}
+        rv = self.server.manager(MGR_CMD_SET, NODE, a, id=self.hostA, expect=True)
+        self.assertTrue(rv)
+        rv = self.server.manager(MGR_CMD_SET, NODE, a, id=self.hostB, expect=True)
+        self.assertTrue(rv)
+
+    def common(self, is_nonrerunnable, restart_mom):
+        """
+        Set node_fail_requeue=5 on server
+        """
+        a = {ATTR_nodefailrq: 5}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        """
+        Job script
+        """
+        test = []
+        test += ['#PBS -N RequeueTest\n']
+        test += ['#PBS -l ncpus=1\n']
+        test += ['echo Starting test at `date`\n']
+        test += ['sleep 1000\n']
+
+        test1 = []
+        test1 += ['#PBS -N RequeueTest\n']
+        test1 += ['#PBS -l nodes=1\n']
+        test1 += ['echo Starting test at `date`\n']
+        test1 += ['sleep 1000\n']
+
+        """
+        Submit three jobs J1,J2,J3[]
+        """
+        j1 = Job(TEST_USER, attrs={ATTR_k: 'oe'})
+        j1.create_script(body=test)
+        jid1 = self.server.submit(j1)
+
+        if is_nonrerunnable==True:
+            j2 = Job(TEST_USER, attrs={ATTR_r: 'n', ATTR_k: 'oe' })
+        else:
+            j2 = Job(TEST_USER, attrs={ATTR_r: 'y', ATTR_k: 'oe'})
+
+        j2.create_script(body=test1)
+        jid2 = self.server.submit(j2)
+
+        j3 = Job(TEST_USER, attrs={ATTR_J: '1-6:1', ATTR_k: 'oe'})
+        j3.create_script(body=test)
+        jid3 = self.server.submit(j3)
+
+        subjobs = self.server.status(JOB, id=jid3, extend='t')
+        jid3s1 = subjobs[1]['id']
+
+        """
+        Wait for the jobs to start running.
+        """
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid1, offset=15)
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid2, offset=15)
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid3s1, offset=15)
+
+        """
+        Verify that accounting logs have Resource_List.<resource> value
+        """
+        m = self.server.accounting_match(
+        msg='.*Resource_List.*', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
```

Comment:
I'd rather see all of the calls be more consistent with other tests.  Each test that gets checked in becomes an example for someone else to follow.  If we aren't consistent, there will be confusion.